doc/mkdocs/docs/api/json_pointer/index.md

```diff
@@ -11,14 +11,18 @@ are the base for JSON patches.
 
 ## Template parameters
 
-`BasicJsonType`
-:   a specialization of [`basic_json`](../basic_json/index.md)
+`RefStringType`
+:   the string type used for the reference tokens making up the JSON pointer
+
+## Notes
+
+For backwards compatibility `RefStringType` may also be a specialization of [`basic_json`](../basic_json/index.md) in which case `string_t` will be deduced as [`basic_json::string_t`](../basic_json/string_t.md). This feature is deprecated and may be removed in a future major version.
```

Comment:
Oh, I didn't know how widely `[[deprecated]]` could be applied. https://en.cppreference.com/w/cpp/language/attributes/deprecated

Hopefully that is true for the other compiler-specific options as well.

Added.

Reply:
So, putting it on the using alias doesn't work on MSVC but it should work on the specialization of `string_t_helper` itself.

Reply:
But accessing `string_t_helper::type` doesn't seem to count as using the type so the warning doesn't trigger. I'm giving up on this unless you or anyone has a suggestion.

Reply:
This doesn't work either on GCC 12/Clang 15. Truly giving up now.
```
    NLOHMANN_BASIC_JSON_TPL_DECLARATION
    struct string_t_helper<NLOHMANN_BASIC_JSON_TPL>
    {
#if JSON_HEDLEY_HAS_ATTRIBUTE(deprecated)
        using type [[deprecated("Since 3.10.6; use basic_json::json_pointer or nlohmann::json_pointer<basic_json::string_t>")]]
            = StringType;
#else
    using type = StringType;
#endif
    };
```